src/sick_tim_common_usb.cpp

```diff
@@ -265,6 +265,7 @@ int SickTimCommonUsb::sendSOPASCommand(const char* request, std::vector<unsigned
   if (device_handle_ == NULL) {
     ROS_ERROR("LIBUSB - device not open");
     diagnostics_.broadcast(diagnostic_msgs::DiagnosticStatus::ERROR, "LIBUSB - device not open.");
+    sleep(1);
```

Comment:
I'd simply put it in the outer loop; that's where it belongs IMO, since we want to add the sleep for rate-limiting purposes of that loop, so it's better to just have it in one place instead of having to make sure that each implementation does it.

True, the TCP implementation would have two distinct delays, but *only* in the first iteration of the loop; `ros::Rate` accounts for the duration between calls, so if there is already a timeout inside the TCP implementation, its length will be subtracted from the time that `ros::Rate` sleeps.